### PR TITLE
feat: set drm type by default

### PR DESCRIPTION
**Pull request by andrewjl-mux**
Set view DRM type to `widevine` when applicable.

### CODE REVIEW

File: library/src/main/java/com/mux/player/MuxPlayer.kt [hunk 1]
```diff
@@ -71,7 +71,20 @@ class MuxPlayer private constructor(
...
+        // TODO: escape hatch TBA
+        if (mediaItem.requestMetadata.extras.getString(Constants.BUNDLE_DRM_TOKEN)) {
+          val viewData = CustomerViewData()
+          // Assumes only widevine DRM playback is supported
+          // If playready support is added in future, update to select between widevine and playready
+          viewData.viewDrmType = "widevine" // TODO: make this a constant
+
+          // TODO: Confirm this doesn't overwrite other keys like view session ID to null
+          // TODO: This is deprecated. Confirm the most current API to use.
+          muxStats?.updateCustomerData(
```
Comment (andrewjl-mux) [inline library/src/main/java/com/mux/player/MuxPlayer.kt:84]: Is there a non-deprecated update method this should be using?

https://github.com/muxinc/mux-stats-sdk-java/blob/452e404b5d52b5ea7a58ca5d140f3860a06b3c01/MuxCore/src/main/java/com/mux/stats/sdk/muxstats/MuxStats.java#L389

File: library/src/main/java/com/mux/player/MuxPlayer.kt [hunk 2]
```diff
@@ -71,7 +71,20 @@ class MuxPlayer private constructor(
...
-        //muxStats?.videoChange(CustomerVideoData())
+        // Check if a DRM token is set, set View Drm Type if it is
+        // TODO: escape hatch TBA
+        if (mediaItem.requestMetadata.extras.getString(Constants.BUNDLE_DRM_TOKEN) != nil) {
+          val viewData = CustomerViewData()
+          // Assumes only widevine DRM playback is supported
+          // If playready support is added in future, update to select between widevine and playready
+          viewData.viewDrmType = Constants.VIEW_DRM_TYPE_WIDEVINE
+
+          // TODO: Confirm this doesn't overwrite other keys like view session ID to null
```
Comment (daytime-em) [inline library/src/main/java/com/mux/player/MuxPlayer.kt:82]: Confirmed, the existing customer data is preserved. There's several hops through the core, but eventually `BaseQueryData.update()` is called, [implementation here](https://github.com/muxinc/mux-stats-sdk-java/blob/452e404b5d52b5ea7a58ca5d140f3860a06b3c01/MuxCore/src/main/java/com/mux/stats/sdk/core/model/BaseQueryData.java#L44).

As you can see, only keys present in the input `.*WhateverData` are set on the existing object.

**Review by daytime-em — CHANGES_REQUESTED** [review]
Had some feedback and some answers for you.

Also, I'd appreciate it if you could make sure the project builds before pushing. If you're having trouble with Android Studio, I can help you get set up.

File: library/src/main/java/com/mux/player/MuxPlayer.kt [hunk 3]
```diff
@@ -138,6 +152,7 @@ class MuxPlayer private constructor(
     private var enableSmartCache: Boolean = false
     private var logger: Logger? = null
     private var customerData: CustomerData = CustomerData()
+    private var didAddMonitoringData: Boolean = false
```
Comment (daytime-em) [inline library/src/main/java/com/mux/player/MuxPlayer.kt:155]: question: The value is never read, was this no longer required?

**Review by daytime-em — COMMENTED** [review]
Just had one question!

**Review by daytime-em — APPROVED** [review]

File: library/src/main/java/com/mux/player/MuxPlayer.kt (hunk 3 above)
Comment (andrewjl-mux) [reply library/src/main/java/com/mux/player/MuxPlayer.kt:155]: It's used here, I think? https://github.com/muxinc/mux-player-android/blob/b95ac515533baf4f3e81a5f69cdc05d5bf6ac853/library/src/main/java/com/mux/player/MuxPlayer.kt#L262

Comment (andrewjl-mux) [reply library/src/main/java/com/mux/player/MuxPlayer.kt:155]: I meant this to work the same way that `MuxPlayer.Builder` `enableSmartCache` does.

Comment (daytime-em) [reply library/src/main/java/com/mux/player/MuxPlayer.kt:155]: ahh, understood! I must have missed that

**Pull request merged**